src/tab/Tab.tsx

```diff
@@ -49,6 +55,9 @@ function Tab({
   );
 
   function handleChangeActiveTab(index: number) {
+    if (onTabChange && index !== activeTabIndex) {
+      onTabChange(index);
+    }
     setActiveTabIndex(index);
```

Comment:
There shouldn't be a state duplication. If it is a controlled component, Tab's own state should never change but the component will be controlled via the prop passed.